tests/lib/rules/no-unnecessary-service-injection-argument.js

```diff
@@ -0,0 +1,68 @@
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-unnecessary-service-injection-argument');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE } = rule;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    sourceType: 'module'
+  }
+});
+
+ruleTester.run('no-unnecessary-service-injection-argument', rule, {
+  valid: [
+    'export default Component.extend({ serviceName: service() });',
+    'export default Component.extend({ serviceName: inject() });',
+
+    "export default Component.extend({ specialName: service('service-name') });",
+    "export default Component.extend({ specialName: inject('service-name') });",
+
+    // With unrecognized second argument:
+    "export default Component.extend({ serviceName: service('serviceName', EXTRA_PROPERTY) });",
+    "export default Component.extend({ serviceName: inject('serviceName', EXTRA_PROPERTY) });",
+
+    // *********************************
+    // Not Ember's `service()` function:
+    // *********************************
+
+    "export default Component.extend({ serviceName: otherFunction('serviceName') });",
+
+    "export default Component.extend({ serviceName: service.otherFunction('serviceName') });",
+    "export default Component.extend({ serviceName: inject.otherFunction('serviceName') });"
+  ],
+  invalid: [
+    {
+      code:
+        "export default Component.extend({ serviceName: service('serviceName') });",
+      output: 'export default Component.extend({ serviceName: service() });',
+      errors: [{ message: ERROR_MESSAGE, type: 'Literal' }]
+    },
+    {
+      code:
+        "export default Component.extend({ serviceName: inject('serviceName') });",
+      output: 'export default Component.extend({ serviceName: inject() });',
+      errors: [{ message: ERROR_MESSAGE, type: 'Literal' }]
+    },
+    {
+      code:
+        "export default Component.extend({ serviceName: service('service-name') });",
+      output: 'export default Component.extend({ serviceName: service() });',
+      errors: [{ message: ERROR_MESSAGE, type: 'Literal' }]
+    },
+    {
+      code:
+        "export default Component.extend({ serviceName: inject('service-name') });",
+      output: 'export default Component.extend({ serviceName: inject() });',
+      errors: [{ message: ERROR_MESSAGE, type: 'Literal' }]
+    }
+  ]
```

Comment:
just in case: can you add an example that uses something other than a Component? and possibly one without the `export default` too?

Reply:
Added more test cases.